### PR TITLE
Make multiple post types

### PR DESCRIPTION
**Pull request by bahiirwa**
Add several post types with options to make support editable in multi-dimentional array and also implement the CPT for REST API implementation ( Gutenberg Friendly :-) ).

### CODE REVIEW

**Comment by BaronVonPerko** [comment]
I like this approach.  

You could also turn each CPT into a Class, where each property of the array is just a property of the class, and load them into the Custom.php file.  This would allow the Custom.php file to be part of the "framework" per say, and you would just build small POPOs for each new CPT you want.

Idk, maybe that is over-engineering it, as what you have is a simple solution.  I'd like to see what the WP expert @Alecaddd has to say.

File: inc/Custom/Custom.php [hunk 1]
```diff
@@ -12,66 +12,87 @@ class Custom
...
-			'query_var'          => true,
-			'rewrite'            => array( 'slug' => 'book' ),
-			'capability_type'    => 'post',
-			'has_archive'        => true,
-			'hierarchical'       => false,
-			'menu_position'      => 5, // below post
-			'supports'           => array( 'title', 'editor', 'author', 'thumbnail', 'excerpt', 'comments' )
+	public function register() {
+		add_action( 'init', array( &$this, 'custom_post_type'), 10 , 4 );
+		add_action( 'after_switch_theme', array( &$this, 'rewrite_flush') );
```
Comment (Alecaddd) [inline inc/Custom/Custom.php]: I like it, good approach.
Any specific reason why you're using the class reference here?

Comment (bahiirwa) [reply inc/Custom/Custom.php]: **Slaps self on head. Oversight!! To answer you and Chris I started my thought process from [CPT-Class](https://github.com/bahiirwa/CPT-Class) that use a full class that is initialized by calling the class. 

@BaronVonPerko does [CPT-Class](https://github.com/bahiirwa/CPT-Class) implement what you were sharing?

@Alecaddd I have made a correction on PR. No need for `&$this`

Comment (BaronVonPerko) [reply inc/Custom/Custom.php]: @bahiirwa I was thinking along the lines of an abstract class that you would create subclasses of describing each CPT.  But your example in CPT-Class would work well as well by instantiating this class and passing it the constructor params needed to build the CPT.  

Both would work fine, just really comes down to how you prefer to write your code :+1:

**Pull request merged**